CMakeLists.txt

```diff
@@ -207,6 +211,20 @@ mark_as_advanced(BUILD_FOR_PACKAGING)
 cmake_dependent_option(FORCE_LEGACY_LIBBPF "Force usage of libbpf 0.0.9 instead of the latest version." False "ENABLE_PLUGIN_EBPF" False)
 mark_as_advanced(FORCE_LEGACY_LIBBPF)
 
+include(CheckFunctionExists)
+
+if(ENABLE_WOLFSSL)
+        pkg_check_modules(TLS IMPORTED_TARGET wolfssl)
+
+        list(APPEND CMAKE_REQUIRED_LIBRARIES wolfssl)
```

Comment:
This needs to be reset after the check so that it doesn’t potentially modify behavior elsewhere. The cleanest option in this case is going to be to define a function to make the check, which will limit the change to the required library list to the scope of the function.